lib/src/Uncertainty/Algorithm/WeightedExperiments/TensorProductExperiment.cxx

```diff
@@ -154,5 +165,17 @@ void TensorProductExperiment::load(Advocate & adv)
   adv.loadAttribute("collection_", collection_);
 }
 
+/* Distribution collection accessor */
+const TensorProductExperiment::WeightedExperimentCollection & TensorProductExperiment::getWeightedExperimentCollection() const
+{
+  return collection_;
+}
+
+/* Distribution collection accessor */
+void TensorProductExperiment::setWeightedExperimentCollection(const WeightedExperimentCollection & coll)
+{
+  isAlreadyComputedNodesAndWeights_ = false;
+  collection_ = coll;
```

Comment:
You should fill `distribution_` and `size_` attributes using `BlockIndependentDistribution([elem.getDistribution() for elem in coll]`) and  product of`elem.getSize()`

Reply:
There is no distribution currently involved in `TensorProductExperiment`. This is because the object is built on top of a collection of experiments. Hence, the user must have created the experiments from the distribution beforehand. 
Would the BlockIndependentDistribution object be available, what could it be used for?

Reply:
`TensorProductExperiment` is a `WeightedExperiment`. As such, it must return the associated distribution when one calls its `getDistribution()` method. And this method is a `BlockIndependentDistribution` built from the collection of `Distribution` embedded into the collection of `WeightedExperiment` given to the constructor.

Reply:
Ok, you point to the same behavior I can see at GaussProductExperiment:

https://github.com/openturns/openturns/blob/b060938db798457a0db5f0b2ac169771a0f8fa4c/lib/src/Uncertainty/Algorithm/WeightedExperiments/GaussProductExperiment.cxx#L139

but tuned to match the information we have at TensorProductExperiment. 
Hence, my unit test must be wrong, because I did not notice that the TensorProductExperiment did not provide the correct methods.

Reply:
Fixed at cfa0997817ebd1527ad5072d1d93075d4dd3be90